enhancements/storage/storage-security-policy.md

```diff
@@ -0,0 +1,198 @@
+
+### Motivation for selinuxChangePolicy defaulting
+
+https://github.com/kubernetes/enhancements/issues/1710 is bringing changes which can be breaking for certain pods which use different selinux contexts for same volume between different pods or same pod.
+
+This will speed up time spent while recursively changing selinux policies of a volume.
+
+By allowing cluster-admins to set namespace default of `selinuxChangePolicy` we want to give control over how these changes are applied to Openshift clusters, without explicitly  changing pods themselves.
+
+### User Stories
+
+N/A
+
+### Goals
+
+- Allow Openshift admins to configure per-namespace default of `fsGroupChangePolicy`.
+- Allow Openshift admins to configure per-namespace default of `selinuxChangePolicy`.
+
+### Non-Goals
+
+* Change Openshift default globally.
+
+## Proposal
+
+### Allow admins to opt-in to `fsGroupChangePolicy` via namespace label
+
+We propose a label `storage.openshift.io/fsgroup-change-policy` if that is set to `OnRootMismatch` in a namespace, then Openshift's SCC hooks start defaulting all pods created in that namespace with `OnRootMismatch` `FSGroupChangePolicy`.
+
+```go
+func getPodFsGroupChangePolicy(ns *corev1.Namespace) *api.PodFSGroupChangePolicy {
+    fsGroupPolicy, ok := ns.Labels[securityv1.FSGroupChangePolicy]
+    if !ok {
+        return nil
+    }
+    if fsGroupPolicy == "OnRootMismatch" {
+        onRootMismatchPolicy := api.FSGroupChangeOnRootMismatch
+        return &onRootMismatchPolicy
+    }
+    return nil
+}
+```
+
+### Allow admins to opt-in to `selinuxChangePolicy` via namespace label
+
+Similar to mechanism proposed above,  we propose to use label `storage.openshift.io/selinux-change-policy` to define namespace wide `selinuxChangePolicy` if none are specified in a pod.
+
+### Setting of `fsGroupChangePolicy` and `selinuxChangePolicy` during admission
+
+We propose to add a new admission hook (which will be eventually replaced with MAP) which will trigger the defautling logic, similar to how SCC matching hooks are triggered.
+
+### Eventual replacement with Mutating Admission policy
+
+We want to replace admission plugin with Mutating Admission Policy once the feature becomes generally available. Once Mutating Admission Policy becomes GA, we propose to replace the admission hook with MAP on the lines of https://docs.google.com/document/d/13xa7imyvQzok3IJ66mheGqemQGhkNySl-He1h1iRuSM/edit?usp=sharing
+
+This would serve similar purpose as admission hook we are carrying as go code.
+
+### Proof of concept
+
+We have already implemented a proof-of-concept https://github.com/openshift/kubernetes/pull/2311/files, which works and does the job. But this uses existing SCC admission hook.
```

Comment:
Have you tried to POC the MAP?

Reply:
I have. Did you had a chance to check Google Doc I have linked in this enhancement?